src/test/java/seedu/address/logic/commands/DeleteAssignmentCommandTest.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalAssignments.getTypicalAssignmentSchedule;
+import static seedu.address.testutil.TypicalIndexes.FIRST_INDEX;
+import static seedu.address.testutil.TypicalIndexes.SECOND_INDEX;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.EventSchedule;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.RestaurantBook;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.assignment.Assignment;
+
+public class DeleteAssignmentCommandTest {
+    private Model model = new ModelManager(new AddressBook(),
+        new RestaurantBook(),
+        getTypicalAssignmentSchedule(),
+        new EventSchedule(),
+        new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Assignment assignmentToDelete = model.getFilteredAssignmentList().get(FIRST_INDEX.getZeroBased());
+        DeleteAssignmentCommand deleteCommand = new DeleteAssignmentCommand(FIRST_INDEX);
+
+        CommandResult expectedCommandResult = new CommandResult(String.format(DeleteAssignmentCommand
+            .MESSAGE_DELETE_ASSIGNMENT_SUCCESS, assignmentToDelete), false, false, false,
+            true, false, false, false, false);
+
+        ModelManager expectedModel = new ModelManager(new AddressBook(),
+            new RestaurantBook(),
+            getTypicalAssignmentSchedule(),
+            new EventSchedule(),
+            new UserPrefs());
+        expectedModel.deleteAssignment(assignmentToDelete);
+
+        assertCommandSuccess(deleteCommand, model, expectedCommandResult, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredAssignmentList().size() + 1);
+        DeleteAssignmentCommand deleteCommand = new DeleteAssignmentCommand(outOfBoundIndex);
+
+        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+    }
+
```

Comment:
Perhaps other than valid and invalid integers, can also consider negative tests where the user enters for example a letter into the INDEX field

Reply:
ermm this one is in the deleteCommandParserTest o;